torchaudio/models/wav2vec2/components.py

```diff
@@ -314,7 +318,185 @@ def forward(
+        relative_position = memory_position - context_position
+        relative_position_bucket = self._relative_positions_bucket(relative_position, bidirectional=True)
+        relative_position_bucket = relative_position_bucket.to(self.rel_attn_embed_device)
+        values = self.rel_attn_embed(relative_position_bucket)
+        values = values.permute([2, 0, 1])
+        return values
+
+    def _relative_positions_bucket(self, relative_positions: Tensor, bidirectional: bool = True):
+        num_buckets = self.num_buckets
+        max_distance = self.max_distance
+        # Shape (query_length, key_length)
+        relative_buckets = torch.zeros_like(relative_positions, dtype=torch.long)
+
+        if bidirectional:
+            num_buckets = num_buckets // 2
+            relative_buckets += (relative_positions > 0).to(torch.long) * num_buckets
+            relative_positions = torch.abs(relative_positions)
+        else:
+            relative_positions = -torch.min(relative_positions, torch.zeros_like(relative_positions))
+
+        max_exact = num_buckets // 2
+        is_small = relative_positions < max_exact
+
+        relative_postion_if_large = max_exact + (
+            torch.log(relative_positions.float() / max_exact)
+            / math.log(max_distance / max_exact)
+            * (num_buckets - max_exact)
+        ).to(torch.long)
+        relative_postion_if_large = torch.min(
+            relative_postion_if_large, torch.full_like(relative_postion_if_large, num_buckets - 1)
+        )
+
+        relative_buckets += torch.where(is_small, relative_positions, relative_postion_if_large)
+        return relative_buckets
+
+    def forward(
+        self,
+        query: Tensor,
+        key_padding_mask: Optional[Tensor] = None,
+        attention_mask: Optional[Tensor] = None,
+        position_bias: Optional[Tensor] = None,
+    ) -> Tuple[Tensor, Optional[Tensor]]:
+        """
+        Args:
+            query: input of shape `(batch_size, src_len, embed_dim)`.
+            key_padding_mask: mask to exclude keys that are pads, of shape `(batch, src_len)`, where padding elements
+                are indicated by 1s.
+            attn_mask: needs to be `None`. The argument exists for compatibility with `EncoderLayer`.
+            position_bias: position bias of shape `(batch_size * num_heads, src_len, src_len)`. When used inside WavLM
+                model encoder, will be generated in the first layer and then passed from each encoder layer
+                to the next one.
+        Returns:
+            attn_output: attention output of shape `(batch_size, src_len, embed_dim)`
+            position_bias: position bias of shape `(batch_size * num_heads, src_len, src_len)`.
+        """
+        bsz, seq_len, embed_dim = query.size()
+        assert embed_dim == self.embed_dim
+        assert attention_mask is None
+
+        if self.has_relative_attention_bias and position_bias is None:
```

Comment:
by changing the conditional logic it can pass the torchscript check.
```suggestion
        if self.rel_attn_embed is not None and position_bias is None:
```